### PR TITLE
chore: add script to list integraitons and plugin files

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Created a shell script to list integration SDK and plugin files in the S3 bucket. It'll generate an HTML file with all the necessary file URLs.
It is integrated directly into the deployment workflows.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-544/have-index-listing-of-integrations-inventory-working-post-s3

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new script to generate an HTML file listing files from an S3 bucket, enhancing component visibility.
- **Documentation**
	- Added details on command-line arguments for the new script to assist users in its implementation.

### CODE REVIEW

File: scripts/list-sdk-components.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    if [[ "$file" == "$excluded" ]]; then
+      return 0 # File is in the excluded list
+    fi
+  done
+  return 1 # File is not in the excluded list
+}
+
+# Get list of files from S3 (non-recursive)
+# Capture the timestamp, file size, and file name for each file
+FILES=$(aws s3 ls s3://$BUCKET_NAME/$DIRECTORY_PATH/ | awk '{print $1, $2, $4}')
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh:31]: Good point but at this point we don't have a lot of files.

File: scripts/list-sdk-components.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  if is_excluded "$FILE"; then
+    continue # Skip this file
+  fi
+
+  # Remove the .min.js or .min.js.map extension for display
+  FILE_DISPLAY=$(basename "$FILE" | sed 's/\.min\.js$//' | sed 's/\.min\.js\.map$//' | sed 's/\\.js$//' | sed 's/\\.js\.map$//')
+
+  if [[ "$FILE" == *.min.js.map || "$FILE" == *.js.map ]]; then
+    # Indent map files
+    echo "<tr><td></td><td style=\"padding-left: 20px;\"><a href=\"$URL\">Source Map</a></td><td>$TIMESTAMP</td></tr>" >> $OUTPUT_HTML_FILE_PATH
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh:62]: We can ignore this. The file names don't contain any special characters.

File: scripts/list-sdk-components.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    COUNTER=$((COUNTER + 1))
+  fi
+done <<< "$FILES"
+
+# Finish HTML file
+echo "</table>" >> $OUTPUT_HTML_FILE_PATH
+echo "</body>" >> $OUTPUT_HTML_FILE_PATH
+echo "</html>" >> $OUTPUT_HTML_FILE_PATH
+
+echo "HTML file with links to S3 files has been generated: $OUTPUT_HTML_FILE_PATH"
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh:75]: It is fine for now.

File: scripts/list-sdk-components.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+OUTPUT_HTML_DIR_PATH=$4
+COMPONENT_NAME=$5
+ZIP_FILE_NAME=$6
+
+OUTPUT_HTML_FILE_PATH="$OUTPUT_HTML_DIR_PATH/$OUTPUT_HTML_FILE_NAME"
+
+# List of files to exclude
+# Exclude the ZIP file and the output HTML file as they'll be
+# in the same directory as the other files
+EXCLUDED_FILES=($ZIP_FILE_NAME $OUTPUT_HTML_FILE)
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh]: Addressed.

File: scripts/list-sdk-components.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  FILE=$(echo $LINE | awk '{print $3}')
+  URL="https://$BUCKET_NAME.s3.amazonaws.com/$DIRECTORY_PATH/$FILE"
+
+  # Check if the file is in the excluded list
+  if is_excluded "$FILE"; then
+    continue # Skip this file
+  fi
+
+  # Remove the .min.js or .min.js.map extension for display
+  FILE_DISPLAY=$(basename "$FILE" | sed 's/\.min\.js$//' | sed 's/\.min\.js\.map$//' | sed 's/\\.js$//' | sed 's/\\.js\.map$//')
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh]: Addressed.

File: scripts/list-sdk-components.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh]: Addressed.

File: scripts/list-sdk-components.sh [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    if [[ "$file" == "$excluded" ]]; then
+      return 0 # File is in the excluded list
+    fi
+  done
+  return 1 # File is not in the excluded list
+}
+
+# Get list of files from S3 (non-recursive)
+# Capture the timestamp, file size, and file name for each file
+FILES=$(aws s3 ls s3://$BUCKET_NAME/$DIRECTORY_PATH/ | awk '{print $1, $2, $4}')
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply scripts/list-sdk-components.sh:45]: Addressed.

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**